### PR TITLE
remove sinon sandbox usage

### PR DESCRIPTION
**Pull request by DanielGluskin**
issue #28

### CODE REVIEW

**Review by goldbergyoni — CHANGES_REQUESTED** [review]
@DanielGluskin This is a great fix.

One important clarification - sinon.restore should exist in beforeEach, otherwise nothing will clean the modifications that test1 is doing and test3 will not be aware of those changes. Sinon has no way to know when a test ends and clean-up. Previously one had to create a sandbox and maintain as global var, now this can be done directly over the sinon object

Try it yourself (you're on your way to be a testing expert:)), make a test stub the 'addOrder' to throw error, see that the next tests also fail.

File: example-application/test/basic-tests.test.js [hunk 1]
```diff
@@ -35,9 +31,7 @@ beforeEach(() => {
     name: "John",
   });
 
-  if (sinonSandbox) {
-    sinonSandbox.restore();
-  }
+  sinon.restore();
```
Comment (mikicho) [inline example-application/test/basic-tests.test.js:34]: IMO this should be in `afterEach`
1. More reasonable.
2. Sometimes you may want to declare some stubs in the `beforeEach` and the restore will remove them

File: various-receipes/authentication/test/auth-example-with-stub.test.js [hunk 2]
```diff
@@ -8,9 +8,9 @@ const {
 const authenticationMiddleware = require("../authentication-middleware");
 
 let expressApp;
-let sinonSandbox;
 
 beforeAll(async (done) => {
+    sinon.restore();
```
Comment (mikicho) [inline various-receipes/authentication/test/auth-example-with-stub.test.js:13]: > Sometimes you may want to declare some stubs in the beforeEach and the restore will remove them

for example... 😅

**Review by mikicho — CHANGES_REQUESTED** [review]
much cleaner!

**Comment by mikicho** [comment]
@DanielGluskin why?

**Comment by DanielGluskin** [comment]
@mikicho was fixed and merged by an other PR.

**Pull request closed without merging**